File: Signup.cs
using System;
using System.Data;
using System.Drawing;
using System.Net;
using System.Net.Mail;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace MilkTeaOrderAndInventorySystem
{

    public partial class Signup
    {
        private DateTime TargetDT;
        private DateTime tDateTime;
        private TimeSpan CountDownFrom = TimeSpan.FromSeconds(60d);
        private TimeSpan stopWatch = TimeSpan.FromSeconds(7d);

        private string randCode;
        private string toUser;

        private readonly byte[] key = new byte[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24 };
        private readonly byte[] iv = new byte[] { 10, 9, 8, 7, 6, 5, 4, 3 };
        private PassEnCDeC des;

        public string sentCode { get; set; }
        public string passEmail { get; set; }

        public Signup()
        {
            des = new PassEnCDeC(key, iv);
            InitializeComponent();
        }

        private void insertData()
        {
            try
            {
                UniversalVariables.sql = "INSERT INTO userProfile (uName, uNCount, uEmail, uPass) VALUES ('" + txtUsername.Text + "', '" + "0" + "' , '" + passEmail + "','" + des.Encrypt(txtPass.Text) + "')";
                UniversalVariables.connect();
                UniversalVariables.conn.Close();

                arrow2.Image = My.Resources.Resources.arrowcolored;
                picCheck.Image = My.Resources.Resources.checkcolored;
                lblDone.ForeColor = Color.FromArgb(190, 65, 109);
                pnlCreate.Visible = false;
                pnlDone.Visible = true;
                timerRedirect();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        public bool ValidatePassword(string pwd, int minLength = 8, int numUpper = 1, int numLower = 5, int numNumbers = 1, int numSpecial = 1)
        {

            // Replace [A-Z] with \p{Lu}, to allow for Unicode uppercase letters.
            var upper = new System.Text.RegularExpressions.Regex("[A-Z]");
            var lower = new System.Text.RegularExpressions.Regex("[a-z]");
            var number = new System.Text.RegularExpressions.Regex("[0-9]");
            // Special is "none of the above".
            var special = new System.Text.RegularExpressions.Regex("[^a-zA-Z0-9]");

            // Check the length.
            if (Strings.Len(pwd) < minLength)
                return false;
            // Check for minimum number of occurrences.
            if (upper.Matches(pwd).Count < numUpper)
                return false;
            if (lower.Matches(pwd).Count < numLower)
                return false;
            if (number.Matches(pwd).Count < numNumbers)
                return false;
            if (special.Matches(pwd).Count < numSpecial)
                return false;

            // Passed all checks.
            return true;
        }

        private bool usernameCheck(string uname)
        {
            bool usernameExist = false;

            try
            {
                UniversalVariables.sql = "SELECT * FROM userProfile WHERE uName = '" + txtUsername.Text + "'";
                UniversalVariables.connect();

                if (UniversalVariables.dr.Read())
                {
                    lblError.Visible = true;
                    lblError.Text = "Username already exist!";
                    usernameExist = true;
                }
                UniversalVariables.conn.Close();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
            return usernameExist;
        }

        private void sendCode()
        {
            string from, pass, messageBody;
            var rand = new Random();
            randCode = rand.Next(999999).ToString();
            var message = new MailMessage();
            toUser = passEmail;
            from = "trialkoito4@gmail.com";
            pass = "broxxedasqosbrvy";
            messageBody = "Your Verification Code is: " + randCode;
            message.To.Add(toUser);
            message.From = new MailAddress(from);
            message.Body = messageBody;
            message.Subject = "Resetting Code";
            var smtp = new SmtpClient("smtp.gmail.com")
            {
                EnableSsl = true,
                Port = 587
            };
            smtp.DeliveryMethod = SmtpDeliveryMethod.Network;
            smtp.Credentials = new NetworkCredential(from, pass);
            try
            {
                smtp.Send(message);

                var obj1 = new CodeVerification();
                obj1.ShowDialog();

                sentCode = randCode;
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        private void timerStarter()
        {
            tmrResend.Start();
            tmrResend.Interval = 100;
            TargetDT = DateTime.Now.Add(CountDownFrom);
        }

        private void timerRedirect()
        {
            tmrRedirect.Start();
            tmrRedirect.Interval = 100;
            tDateTime = DateTime.Now.Add(stopWatch);
        }

        private void btnBack_Click(object sender, EventArgs e)
        {
            var obj = new Login();
            obj.Show();
            Hide();
        }

        private void Signup_Load(object sender, EventArgs e)
        {
            if (UniversalVariables.conn.State == ConnectionState.Open)
            {
                UniversalVariables.conn.Close();
            }
            UniversalVariables.conn.ConnectionString = "server=localhost;user id=root;port=3306;password=0102bryan;database=accountdb";

            txtCode.MaxLength = 6;
            lblEmail.Text = passEmail;
            lblDEmail.Text = passEmail;
            timerStarter();
        }

        private void txtCode_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (!char.IsNumber(e.KeyChar) && !char.IsControl(e.KeyChar))
                e.KeyChar = Conversions.ToChar("");
        }

        private void btnVerify_Click(object sender, EventArgs e)
        {
            if (txtCode.Text.Equals(sentCode))
            {
                arrow1.Image = My.Resources.Resources.arrowcolored;
                pic2.Image = My.Resources.Resources.colored;
                lblCreate.ForeColor = Color.FromArgb(190, 65, 109);

                pnlVerify.Visible = false;
                pnlCreate.Visible = true;
            }
            else
            {
                lblCodeError.Visible = true;
            }
        }

        private void Timer1_Tick(object sender, EventArgs e)
        {
            var ts = TargetDT.Subtract(DateTime.Now);
            if (ts.TotalMilliseconds > 1d)
            {
                lblTimer.Text = "Please wait " + ts.ToString("ss") + " seconds to resend.";
            }
            else
            {
                tmrResend.Stop();
                lblTimer.Visible = false;
                lblQuestion.Visible = true;
                lblResend.Visible = true;
            }
        }

        private void lblResend_Click(object sender, EventArgs e)
        {
            sendCode();
            timerStarter();
            lblTimer.Visible = true;
            lblQuestion.Visible = false;
            lblResend.Visible = false;
        }

        private void Guna2TextBox1_TextChanged(object sender, EventArgs e)
        {
            lblError.Visible = false;

            lblUsername.Visible = true;
            if (string.IsNullOrEmpty(txtUsername.Text))
            {
                lblUsername.Visible = false;
            }
        }

        private void Guna2TextBox2_TextChanged(object sender, EventArgs e)
        {
            lblError.Visible = false;

            lblPass.Visible = true;
            if (string.IsNullOrEmpty(txtPass.Text))
            {
                lblPass.Visible = false;
            }
        }

        private void txtCPass_TextChanged(object sender, EventArgs e)
        {
            lblError.Visible = false;

            lblCPass.Visible = true;
            if (string.IsNullOrEmpty(txtCPass.Text))
            {
                lblCPass.Visible = false;
            }
        }

        private void showPass_CheckedChanged(object sender, EventArgs e)
        {
            if (showPass.Checked == true)
            {
                txtPass.UseSystemPasswordChar = false;
            }
            else
            {
                txtPass.UseSystemPasswordChar = true;
            }
        }

        private void showCPass_CheckedChanged(object sender, EventArgs e)
        {
            if (showCPass.Checked == true)
            {
                txtCPass.UseSystemPasswordChar = false;
            }
            else
            {
                txtCPass.UseSystemPasswordChar = true;
            }
        }

        private void txtCode_TextChanged(object sender, EventArgs e)
        {
            lblCodeError.Visible = false;

            if (txtCode.TextLength == 6)
            {
                btnVerify.Enabled = true;
            }
            else
            {
                btnVerify.Enabled = false;
            }
        }

        private void btnSignup_Click(object sender, EventArgs e)
        {
            lblError.Location = new Point(86, 254);

            if (string.IsNullOrEmpty(txtUsername.Text) | string.IsNullOrEmpty(txtPass.Text) | string.IsNullOrEmpty(txtCPass.Text))
            {
                lblError.Visible = true;
                lblError.Text = "Please fill out the form!";
            }
            else if (txtUsername.Text.Contains(" "))
            {
                lblError.Visible = true;
                lblError.Text = "Username should not contain spaces!";
            }
            else if (txtPass.TextLength < 8)
            {
                lblError.Visible = true;
                lblError.Text = "Password should be at least 8 characters long!";
            }
            else if (txtPass.Text.Equals(txtCPass.Text))
            {
                if (!usernameCheck(txtUsername.Text))
                {
                    if (ValidatePassword(txtPass.Text))
                    {
                        insertData();
                    }
                    else
                    {
                        lblError.Visible = true;
                        lblError.Location = new Point(15, 254);
                        lblError.Text = "Password must contain at least 1 Uppercase, 1 Number, and 1 Special!";
                    }
                }
            }
            else
            {
                lblError.Visible = true;
                lblError.Text = "Password does not match!";
            }
        }

        private void tmrRedirect_Tick(object sender, EventArgs e)
        {
            var ts = tDateTime.Subtract(DateTime.Now);
            if (ts.TotalMilliseconds > 1d)
            {
                lblRedirect.Text = "You will be redirected to Login in " + ts.ToString("ss") + " seconds.";
            }
            else
            {
                tmrRedirect.Stop();
                var obj = new Login();
                obj.signupPanel.Visible = false;
                obj.loginPanel.Visible = true;
                obj.Show();
                Hide();
            }
        }

        private void btnDone_Click(object sender, EventArgs e)
        {
            var obj = new Login();
            obj.Show();
            Hide();
        }

        private void picBack_Click(object sender, EventArgs e)
        {
            pnlCreate.Visible = false;
            pnlVerify.Visible = true;
            txtCode.Text = "";

            pic2.Image = My.Resources.Resources.circle2;
            arrow1.Image = My.Resources.Resources.arrowgray;
            lblCreate.ForeColor = SystemColors.ActiveBorder;
            timerStarter();
        }
    }
}

File: Signup.vb
Imports System.Net
Imports System.Net.Mail

Public Class Signup
    Private TargetDT As DateTime
    Private tDateTime As DateTime
    Private CountDownFrom As TimeSpan = TimeSpan.FromSeconds(60)
    Private stopWatch As TimeSpan = TimeSpan.FromSeconds(7)

    Dim randCode As String
    Dim toUser As String

    Private ReadOnly key() As Byte = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24}
    Private ReadOnly iv() As Byte = {10, 9, 8, 7, 6, 5, 4, 3}
    Dim des As New PassEnCDeC(key, iv)

    Public Property sentCode As String
    Public Property passEmail As String

    Private Sub insertData()
        Try
            sql = "INSERT INTO userProfile (uName, uNCount, uEmail, uPass) VALUES ('" & txtUsername.Text & "', '" & "0" & "' , '" & passEmail & "','" & des.Encrypt(txtPass.Text) & "')"
            connect()
            conn.Close()

            arrow2.Image = My.Resources.arrowcolored
            picCheck.Image = My.Resources.checkcolored
            lblDone.ForeColor = Color.FromArgb(190, 65, 109)
            pnlCreate.Visible = False
            pnlDone.Visible = True
            timerRedirect()
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
    End Sub

    Function ValidatePassword(ByVal pwd As String,
        Optional ByVal minLength As Integer = 8,
        Optional ByVal numUpper As Integer = 1,
        Optional ByVal numLower As Integer = 5,
        Optional ByVal numNumbers As Integer = 1,
        Optional ByVal numSpecial As Integer = 1) As Boolean

        ' Replace [A-Z] with \p{Lu}, to allow for Unicode uppercase letters.
        Dim upper As New System.Text.RegularExpressions.Regex("[A-Z]")
        Dim lower As New System.Text.RegularExpressions.Regex("[a-z]")
        Dim number As New System.Text.RegularExpressions.Regex("[0-9]")
        ' Special is "none of the above".
        Dim special As New System.Text.RegularExpressions.Regex("[^a-zA-Z0-9]")

        ' Check the length.
        If Len(pwd) < minLength Then Return False
        ' Check for minimum number of occurrences.
        If upper.Matches(pwd).Count < numUpper Then Return False
        If lower.Matches(pwd).Count < numLower Then Return False
        If number.Matches(pwd).Count < numNumbers Then Return False
        If special.Matches(pwd).Count < numSpecial Then Return False

        ' Passed all checks.
        Return True
    End Function

    Private Function usernameCheck(ByVal uname As String) As Boolean
        Dim usernameExist As Boolean = False

        Try
            sql = "SELECT * FROM userProfile WHERE uName = '" & txtUsername.Text & "'"
            connect()

            If dr.Read Then
                lblError.Visible = True
                lblError.Text = "Username already exist!"
                usernameExist = True
            End If
            conn.Close()
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
        Return usernameExist
    End Function

    Private Sub sendCode()
        Dim from, pass, messageBody As String
        Dim rand As New Random()
        randCode = (rand.Next(999999)).ToString()
        Dim message As New MailMessage()
        toUser = passEmail
        from = "trialkoito4@gmail.com"
        pass = "broxxedasqosbrvy"
        messageBody = "Your Verification Code is: " + randCode
        message.To.Add(toUser)
        message.From = New MailAddress(from)
        message.Body = messageBody
        message.Subject = "Resetting Code"
        Dim smtp As New SmtpClient("smtp.gmail.com") With {
            .EnableSsl = True,
            .Port = 587
        }
        smtp.DeliveryMethod = smtp.DeliveryMethod.Network
        smtp.Credentials = New NetworkCredential(from, pass)
        Try
            smtp.Send(message)

            Dim obj1 As New CodeVerification
            obj1.ShowDialog()

            sentCode = randCode
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
    End Sub

    Private Sub timerStarter()
        tmrResend.Start()
        tmrResend.Interval = 100
        TargetDT = DateTime.Now.Add(CountDownFrom)
    End Sub

    Private Sub timerRedirect()
        tmrRedirect.Start()
        tmrRedirect.Interval = 100
        tDateTime = DateTime.Now.Add(stopWatch)
    End Sub

    Private Sub btnBack_Click(sender As Object, e As EventArgs) Handles btnBack.Click
        Dim obj As New Login
        obj.Show()
        Me.Hide()
    End Sub

    Private Sub Signup_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        If conn.State = ConnectionState.Open Then
            conn.Close()
        End If
        conn.ConnectionString = "server=localhost;user id=root;port=3306;password=0102bryan;database=accountdb"

        txtCode.MaxLength = 6
        lblEmail.Text = passEmail
        lblDEmail.Text = passEmail
        timerStarter()
    End Sub

    Private Sub txtCode_KeyPress(sender As Object, e As KeyPressEventArgs) Handles txtCode.KeyPress
        If Not Char.IsNumber(e.KeyChar) AndAlso Not Char.IsControl(e.KeyChar) Then e.KeyChar = ""
    End Sub

    Private Sub btnVerify_Click(sender As Object, e As EventArgs) Handles btnVerify.Click
        If txtCode.Text.Equals(sentCode) Then
            arrow1.Image = My.Resources.arrowcolored
            pic2.Image = My.Resources.colored
            lblCreate.ForeColor = Color.FromArgb(190, 65, 109)

            pnlVerify.Visible = False
            pnlCreate.Visible = True
        Else
            lblCodeError.Visible = True
        End If
    End Sub

    Private Sub Timer1_Tick(sender As Object, e As EventArgs) Handles tmrResend.Tick
        Dim ts As TimeSpan = TargetDT.Subtract(DateTime.Now)
        If ts.TotalMilliseconds > 1 Then
            lblTimer.Text = "Please wait " & ts.ToString("ss") & " seconds to resend."
        Else
            tmrResend.Stop()
            lblTimer.Visible = False
            lblQuestion.Visible = True
            lblResend.Visible = True
        End If
    End Sub

    Private Sub lblResend_Click(sender As Object, e As EventArgs) Handles lblResend.Click
        sendCode()
        timerStarter()
        lblTimer.Visible = True
        lblQuestion.Visible = False
        lblResend.Visible = False
    End Sub

    Private Sub Guna2TextBox1_TextChanged(sender As Object, e As EventArgs) Handles txtUsername.TextChanged
        lblError.Visible = False

        lblUsername.Visible = True
        If txtUsername.Text = Nothing Then
            lblUsername.Visible = False
        End If
    End Sub

    Private Sub Guna2TextBox2_TextChanged(sender As Object, e As EventArgs) Handles txtPass.TextChanged
        lblError.Visible = False

        lblPass.Visible = True
        If txtPass.Text = Nothing Then
            lblPass.Visible = False
        End If
    End Sub

    Private Sub txtCPass_TextChanged(sender As Object, e As EventArgs) Handles txtCPass.TextChanged
        lblError.Visible = False

        lblCPass.Visible = True
        If txtCPass.Text = Nothing Then
            lblCPass.Visible = False
        End If
    End Sub

    Private Sub showPass_CheckedChanged(sender As Object, e As EventArgs) Handles showPass.CheckedChanged
        If showPass.Checked = True Then
            txtPass.UseSystemPasswordChar = False
        Else
            txtPass.UseSystemPasswordChar = True
        End If
    End Sub

    Private Sub showCPass_CheckedChanged(sender As Object, e As EventArgs) Handles showCPass.CheckedChanged
        If showCPass.Checked = True Then
            txtCPass.UseSystemPasswordChar = False
        Else
            txtCPass.UseSystemPasswordChar = True
        End If
    End Sub

    Private Sub txtCode_TextChanged(sender As Object, e As EventArgs) Handles txtCode.TextChanged
        lblCodeError.Visible = False

        If txtCode.TextLength = 6 Then
            btnVerify.Enabled = True
        Else
            btnVerify.Enabled = False
        End If
    End Sub

    Private Sub btnSignup_Click(sender As Object, e As EventArgs) Handles btnSignup.Click
        lblError.Location = New Point(86, 254)

        If txtUsername.Text = Nothing Or txtPass.Text = Nothing Or txtCPass.Text = Nothing Then
            lblError.Visible = True
            lblError.Text = "Please fill out the form!"
        ElseIf txtUsername.Text.Contains(" ") Then
            lblError.Visible = True
            lblError.Text = "Username should not contain spaces!"
        ElseIf txtPass.TextLength < 8 Then
            lblError.Visible = True
            lblError.Text = "Password should be at least 8 characters long!"
        Else
            If txtPass.Text.Equals(txtCPass.Text) Then
                If Not usernameCheck(txtUsername.Text) Then
                    If ValidatePassword(txtPass.Text) Then
                        insertData()
                    Else
                        lblError.Visible = True
                        lblError.Location = New Point(15, 254)
                        lblError.Text = "Password must contain at least 1 Uppercase, 1 Number, and 1 Special!"
                    End If
                End If
            Else
                lblError.Visible = True
                lblError.Text = "Password does not match!"
            End If
        End If
    End Sub

    Private Sub tmrRedirect_Tick(sender As Object, e As EventArgs) Handles tmrRedirect.Tick
        Dim ts As TimeSpan = tDateTime.Subtract(DateTime.Now)
        If ts.TotalMilliseconds > 1 Then
            lblRedirect.Text = "You will be redirected to Login in " & ts.ToString("ss") & " seconds."
        Else
            tmrRedirect.Stop()
            Dim obj As New Login
            obj.signupPanel.Visible = False
            obj.loginPanel.Visible = True
            obj.Show()
            Me.Hide()
        End If
    End Sub

    Private Sub btnDone_Click(sender As Object, e As EventArgs) Handles btnDone.Click
        Dim obj As New Login
        obj.Show()
        Me.Hide()
    End Sub

    Private Sub picBack_Click(sender As Object, e As EventArgs) Handles picBack.Click
        pnlCreate.Visible = False
        pnlVerify.Visible = True
        txtCode.Text = ""

        pic2.Image = My.Resources.circle2
        arrow1.Image = My.Resources.arrowgray
        lblCreate.ForeColor = SystemColors.ActiveBorder
        timerStarter()
    End Sub
End Class